src/Compilers/Core/Portable/NativePdbWriter/PdbWriter.cs

```diff
@@ -620,8 +682,31 @@ public void SetMetadataEmitter(MetadataWriter metadataWriter)
             }
         }
 
-        public unsafe void GetDebugDirectoryGuidAndStampAndAge(out Guid guid, out uint stamp, out uint age)
+        public unsafe ContentId GetContentId()
         {
+            if (_deterministic)
+            {
+                // Call to GetDebugInfo fails for SymWriter initialized using InitializeDeterministic.
+                // We already have all the info we need though.
+
+                // TODO: calculate sha1 hash
```

Comment:
See also https://github.com/gafter/roslyn/pull/3 which is a start on the code that will fill in the hole here.